internal/services/elastic/elastic.go

```diff
@@ -98,6 +101,104 @@ func (g *GetHistoryParams) SetDefaultHistoryParams() {
 	}
 }
 
+func (s *Service) GetTotalDailyDistanceDriven(ctx context.Context, tz, deviceID string) ([]byte, error) {
+	query := &search.Request{
+		Query: &types.Query{
+			Bool: &types.BoolQuery{
+				Filter: []types.Query{
+					{Match: map[string]types.MatchQuery{"subject": {Query: deviceID}}},
+					{Range: map[string]types.RangeQuery{"data.timestamp": types.DateRangeQuery{Gte: some.String("now-13d/d"), TimeZone: &tz}}},
+				},
+			},
+		},
+		Size: some.Int(0),
+		Aggregations: map[string]types.Aggregations{
+			"days": {
+				DateHistogram: &types.DateHistogramAggregation{
+					Field:            some.String("data.timestamp"),
+					CalendarInterval: &calendarinterval.Day,
+					TimeZone:         &tz,
+				},
+				Aggregations: map[string]types.Aggregations{
+					"min_odom": {
+						Min: &types.MinAggregation{
+							Field: some.String("data.odometer"),
+						},
+					},
+					"max_odom": {
+						Max: &types.MaxAggregation{
+							Field: some.String("data.odometer"),
+						},
+					},
+					// Code generation for buckets_path is broken as of 8.5.0
+					// See https://github.com/elastic/go-elasticsearch/issues/570
+				},
+			},
+		},
+	}
+
+	resp, err := s.esClient.Search().Index(s.settings.DeviceDataIndexName).Request(query).Perform(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close() // nolint
+
+	if c := resp.StatusCode; c != 200 {
+		s.log.Error().Int("statusCode", c).Msg("Failed to get daily distance from Elastic.")
+		return nil, fmt.Errorf("got status code %d from Elastic message", c)
```

Comment:
We do not need to log and return the error since the error handler will log this for us. 
https://github.com/DIMO-Network/device-data-api/blob/8a826ebd1a15ec0921dcd564001763162dfcadeb/cmd/device-data-api/api.go#L171